lint_rules/lint_rules/debug_rules.py

```diff
@@ -183,9 +183,22 @@ def check_subroutine(cls, subroutine, rule_report, config, **kwargs):
                 dummy_size = Product(dummy_arg_size)
                 stat = cls.compare_sizes(arg_size, alt_arg_size, dummy_size)
 
-                # if necessary, update dimension names and check
-                if not stat:
-                    dummy_size = Product(SubstituteExpressions(assign_map).visit(dummy_arg_size))
+                # we check for a maximum of two indirections for the dummy and arg dimension names
+                for _ in range(2):
```

Comment:
Can we make this indirection depth a config value? That would allow overwriting the value from outside if necessary. See for example https://github.com/ecmwf-ifs/loki/blob/5ed914c00d371fb356889aa457118fe1de0502f3/lint_rules/lint_rules/ifs_coding_standards_2011.py#L36-L38

Reply:
Aah that is a great idea! Please wait to merge until I've added this.